addons/pos_loyalty/models/pos_session.py

```diff
@@ -84,73 +62,24 @@ def _replace_ilike_with_in(self, domain_str):
 
         return json.dumps(domain)
 
-    def _get_pos_ui_product_product(self, params):
-        result = super()._get_pos_ui_product_product(params)
-        self = self.with_context(**params['context'])
-        rewards = self.config_id._get_program_ids().reward_ids
-        products = rewards.discount_line_product_id | rewards.reward_product_ids
-        products |= self.config_id._get_program_ids().filtered(lambda p: p.program_type == 'ewallet').trigger_product_ids
-        # Only load products that are not already in the result
-        products = list(set(products.ids) - set(product['id'] for product in result))
-        products = self.env['product.product'].search_read([('id', 'in', products)], fields=params['search_params']['fields'])
-        self._process_pos_ui_product_product(products)
-        result.extend(products)
-        return result
+    def load_data(self, models_to_load, only_data=False):
```

Comment:
I'm not sure why but the trigger products are not loaded when we start pos with restricted categories.
https://youtu.be/LEVVhS6upow